content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/StateMachines/Job.class/__methods__/launch_ansible_job.rb

```diff
@@ -30,7 +30,7 @@ def main
 
                 def target
                   vm = @handle.root['vm'] || vm_from_request
-                  vm.name if vm
+                  vm.try(:floating_ip) || vm.try(:ipaddresses).first if vm
```

Comment:
This line is not doing what you think it is.  The property `floating_ip` or even `floating_ips` is returning an `FloatingIP` instance which you would then have to call `.address` on to get the actual IP address.
See https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/cloud_manager/vm.rb#L11-L14

Instead you want to reference the virtual columns `ipaddresses` and `floating_ip_addresses` defined here which are arrays of IP address strings: https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/cloud_manager/vm.rb#L32-L33

I would suggest you follow the logic already defined here which returns the IP address string to use:
https://github.com/ManageIQ/manageiq-content/blob/master/content/automate/ManageIQ/System/Request.class/__methods__/order_ansible_playbook.rb#L45-L52

Reply:
@gmcculloug Thanks for bringing that to my attention.  Please let me know if there is anything else you would like changed.